vendor/bat-native-ledger/src/bat/ledger/internal/media/reddit.cc

```diff
@@ -282,4 +282,91 @@ std::string Reddit::GetProfileImageUrl(const std::string& response) {
   return image_url;  // old reddit does not use account icons
 }
 
+void Reddit::OnMediaPublisherInfo(
+    const std::string& user_name,
+    ledger::PublisherInfoCallback callback,
+    ledger::Result result,
+    ledger::PublisherInfoPtr publisher_info) {
+  if (result != ledger::Result::LEDGER_OK &&
+      result != ledger::Result::NOT_FOUND) {
+    callback(ledger::Result::LEDGER_ERROR, nullptr);
+    return;
+  }
+  GURL url(REDDIT_USER_URL + ledger_->URIEncode(user_name));
+  if (!url.is_valid()) {
+    callback(ledger::Result::TIP_ERROR, std::move(publisher_info));
+    return;
+  }
+
+  if (!publisher_info || result == ledger::Result::NOT_FOUND) {
+    FetchDataFromUrl(url.spec(),
+        std::bind(&Reddit::OnPageDataFetched,
+          this,
+          user_name,
+          callback,
+          _1,
+          _2,
+          _3));
+  } else {
+    callback(result, std::move(publisher_info));
+  }
+}
+
+void Reddit::SavePublisherInfo(
+    const std::string& user_name,
+    ledger::PublisherInfoCallback callback,
+    const std::string& data) {
+  const std::string user_id = GetUserId(data);
+  const std::string publisher_key = GetPublisherKey(user_id);
+  const std::string url = GetProfileUrl(user_name);
+  const std::string favicon_url = GetProfileImageUrl(data);
+  const std::string media_key = GetMediaKey(user_name, REDDIT_MEDIA_TYPE);
+
+  if (publisher_key.empty()) {
```

Comment:
nit: maybe try to exit as soon as possible

So when we get `publisher_key` on line 320 we check if valid, before continue parsing other things

Reply:
done